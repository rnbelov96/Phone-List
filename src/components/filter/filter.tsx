import React from 'react';
import { connect } from 'react-redux';
import { AppActionCreators } from '../../reducer/app/app';

export type FilterProps = {
  albumIdList: number[];

  setIdFilter: (id: number) => void;
};

export const PureFilter: React.FunctionComponent<FilterProps> = ({
  albumIdList,
  setIdFilter,
}: FilterProps) => (
  <section className="filter">
    <p className="filter__text">Filter by id:</p>
    <select
      onChange={e => {
        setIdFilter(Number(e.target.value));
      }}
      className="form-select filter__select"
    >
      <option value={0}>All</option>
      {albumIdList.map(id => (
        <option key={id} value={id}>{id}</option>
      ))}
    </select>
  </section>
);

export default connect(() => ({}), {
  setIdFilter: AppActionCreators.setIdFilter,
})(PureFilter);
