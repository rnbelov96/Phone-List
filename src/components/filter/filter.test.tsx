import React from 'react';
import renderer from 'react-test-renderer';
import { PureFilter } from './filter';

test('Modal should render correctly', () => {
  const tree = renderer
    .create(
      <PureFilter
        setIdFilter={() => {}}
        albumIdList={[1, 2, 3]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
