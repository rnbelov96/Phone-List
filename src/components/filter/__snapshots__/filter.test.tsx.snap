// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
<section
  className="filter"
>
  <p
    className="filter__text"
  >
    Filter by id:
  </p>
  <select
    className="form-select filter__select"
    onChange={[Function]}
  >
    <option
      value={0}
    >
      All
    </option>
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
    <option
      value={3}
    >
      3
    </option>
  </select>
</section>
`;
